unidler.py

```diff
@@ -6,36 +7,74 @@
 import ssl
 import sys
 
-# import kubernetes
-# import requests
+import kubernetes
+from kubernetes import client, config
+from kubernetes.client.models import (
+    V1beta1HTTPIngressPath,
+    V1beta1HTTPIngressRuleValue,
+    V1beta1IngressBackend,
+    V1beta1IngressRule,
+)
+
+
+IDLED = 'mojanalytics.xyz/idled'
+IDLED_AT = 'mojanalytics.xyz/idled-at'
+INGRESS_CLASS = 'kubernetes.io/ingress.class'
+UNIDLER = 'unidler'
 
 
 logging.basicConfig(level=os.environ.get('LOG_LEVEL', 'DEBUG'))
 log = logging.getLogger('unidler')
 
 
-class RequestHandler(BaseHTTPRequestHandler):
+def run(host='0.0.0.0', port=8080):
+    try:
+        config.load_incluster_config()
+    except:
+        config.load_kube_config()
 
-    def do_HEAD(self):
-        self.respond(200)
+    unidler = UnidlerServer((host, port), RequestHandler)
+    log.info(f'Unidler listening on {host}:{port}')
+    unidler.serve_forever()
+
+
+class UnidlerServer(ThreadingMixIn, HTTPServer):
+    pass
+
+
+class RequestHandler(BaseHTTPRequestHandler):
 
     def do_GET(self):
-        log.debug(self.requestline)
-        log.debug(self.headers)
+        hostname = self.headers.get('X-Forwarded-Host')
+
+        if not hostname:
+            return self.respond(200, 'OK')
 
-        self.cache_request()
-        self.mark_unidling()
-        # self.restore_replicas()
-        # self.restore_ingress()
-        # self.unmark_idled()
+        log.info(f'Received {self.requestline} for host {hostname}')
 
-        self.respond(200, 'OK')
+        try:
+            unidle_deployment(hostname)
+            log.info('Unidled {hostname}')
+            url = f'//{hostname}{self.path}'
+            body = f'Unidling, please <a href="{url}">try again</a>'
+            self.respond(503, body, {
```

Comment:
It may be nice to use constants for the status codes, maybe even [use `http.HTTPStatus`](https://docs.python.org/3/library/http.html#http.HTTPStatus) unless you think it's overkill - just an idea, not a problem.

Reply:
👍